sapirbitan/assignment2/ex1.cpp

```diff
@@ -0,0 +1,136 @@
+#include <iostream>
+using namespace std;
+
+struct node {
+	int key;
+	node* left_child = nullptr;
```

Comment:
[optional] If you have some time, consider learning about [std::unique_ptr](http://www.drdobbs.com/cpp/c11-uniqueptr/240002708) and how to use it instead of a raw pointers you are currently using. Might be useful for future assignments :)

Your current code works because you never explicitly allocate memory for your data structures, but as soon as you start doing it, i.e. writing something like `node* left_child = new node();` you're in a danger zone for memory leaks and unique_ptrs are a must :)